### PR TITLE
Add class resolver for hubness reduction methods

### PR DESCRIPTION
**Pull request by cthoyt**
This PR uses the `class_resolver` to make it easier to specify different hubness reducers. In addition to being able to pass a pre-instantiated hubness reducers, the following are now possible:

## Demo

Boilerplate code:

```python
import numpy as np

from kiez import Kiez
from kiez.neighbors import HNSW
from kiez.hubness_reduction import CSLS

rng = np.random.RandomState(0)
source, target = rng.rand(100,50), rng.rand(100,50)
```

### With class

```python
hnsw = HNSW(n_candidates=10)
k_inst = Kiez(n_neighbors=5, algorithm=hnsw, hubness=CSLS)
...
```

### With class and kwargs

```python
hnsw = HNSW(n_candidates=10)
k_inst = Kiez(n_neighbors=5, algorithm=hnsw, hubness=CSLS, hubness_kwargs=dict(k=4))
...
```

### With string

```python
hnsw = HNSW(n_candidates=10)
k_inst = Kiez(n_neighbors=5, algorithm=hnsw, hubness='CSLS')
...
```

### With string and kwargs

```python
hnsw = HNSW(n_candidates=10)
k_inst = Kiez(n_neighbors=5, algorithm=hnsw, hubness='CSLS', hubness_kwargs=dict(k=4))
...
```

## Why?

Why make this more complicated? Because people want to use ML with JSON/YAML configs, and those will have to contain string information about the names of classes. `class_resolver` makes this all work. Also, you don't have to import as much stuff.

## Future

With a little more thought, this could also be applied to the algorithm as well, then you could completely load up a Kiez instance directly from JSON like

```python
kiez_kwargs = dict(
    n_neighbors=5,
    algorithm='HNSW',
    hubness='CSLS',
    hubness_kwargs=dict(k=4),
)
kiez = Kiez(**kiez_kwargs)
...
```

### CODE REVIEW

**Comment by cthoyt** [comment]
@dobraczka I'm not familiar with poetry - somehow this PR needs the addition of `class_resolver` as a dependency before it will work (it's a very light package with minimal requirements, almost pure python)

File: kiez/kiez.py [hunk 1]
```diff
@@ -160,7 +167,7 @@ def kneighbors(
         source_query_points=None,
         k=None,
         return_distance=True,
-    ) -> Union[np.ndarray, Tuple(np.ndarray, np.ndarray)]:
+    ) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
```
Comment (cthoyt) [inline kiez/kiez.py:170]: this isn't related to the PR, but it needed to be fixed to pass CI

**Comment by dobraczka** [comment]
Thanks for the contribution, this is very neat!

File: kiez/kiez.py (hunk 1 above)
Comment (dobraczka) [reply kiez/kiez.py:170]: I think it is because you deleted the imports for Tuple and Any before

**Comment by dobraczka** [comment]
I just noticed, that CI is now failing, because I created a custom test badge creation workflow where I forgot, that this would not work if somebody else runs it.. Sorry, I'll just remove that for now and refactor that after merging.

File: kiez/kiez.py (hunk 1 above)
Comment (cthoyt) [reply kiez/kiez.py:170]: type annotations require square bracket syntax and not round brackets

**Pull request merged**